_deploy/r/gnoswap/gns/halving.gno

```diff
@@ -241,3 +250,23 @@ func getYearlyBlockForTimestampMs(ms int64) int64 {
 	yearMs := TIMESTAMP_365DAYS * 1000 // how much millisecond in a year
 	return yearMs / ms                 // how many block in a year
 }
+
+func origCaller() string {
+	return std.GetOrigCaller().String()
+}
+
+func prevRealm() string {
+	return std.PrevRealm().PkgPath()
+}
+
+func callType() string {
+	if prevRealm() == "" {
+		return "DIRECT"
+	}
+
+	return "INDIRECT"
+}
+
+func int64ToStr(i int64) string {
+	return strconv.FormatInt(i, 10)
+}
```

Comment:
It seems to be an unused function. Also, I don't think it needs to be made into a separate function
```suggestion
```

Reply:
It's being used in halving.gno L#180

Reply:
I didn't see that in Emit 👍